lib/asyncify.js

```diff
@@ -79,6 +78,10 @@ export default function asyncify(func) {
     });
 }
 
+function isObject(value) {
+    return value && (typeof value === 'object' || typeof value === 'function');
+}
```

Comment:
Perhaps we can drop this function and use:

```diff
-  if (isObject(result) && typeof result.then === 'function') {
+  if (result && typeof result.then === 'function') {
```

Reply:
That's reasonable, matches other `isThennable` implementations I've seen.